cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -12,198 +12,632 @@
+                        effect=Effect.ALLOW,
+                        resources=[
+                            self.format_arn(
+                                service="s3",
+                                account=None,  # No account ID needed here
+                                region=None,  # No region needed here
+                                resource="{0}/{1}/batch/".format(
+                                    self._cluster.cluster_s3_bucket, self._cluster.artifacts_s3_root_directory
+                                ),
+                            )
+                        ],
+                    ),
+                    PolicyStatement(
+                        sid="FSx", actions=["fsx:DescribeFileSystems"], effect=Effect.ALLOW, resources=["*"]
+                    ),
+                    PolicyStatement(
+                        sid="BatchJobPassRole",
+                        actions=["iam:PassRole"],
+                        effect=Effect.ALLOW,
+                        resources=[self.format_arn(service="iam", resource="role/parallelcluster-*")],
+                    ),
+                    PolicyStatement(
+                        sid="DcvLicense",
+                        actions=["s3:GetObject"],
+                        effect=Effect.ALLOW,
+                        resources=[
+                            self.format_arn(
+                                service="s3",
+                                account=None,  # No account ID needed here
+                                region=None,  # No region needed here
+                                resource="dcv-license.{0}/*".format(self.region),
+                            )
+                        ],
+                    ),
+                ]
+            ),
+            roles=[self.root_iam_role.ref],
         )
 
-        # TODO metadata?
-
-        # https://docs.aws.amazon.com/cdk/latest/guide/use_cfn_template.html
-        # with open('master-server-substack.cfn.yaml', 'r') as f:
-        # template = yaml.load(f, Loader=yaml.SafeLoader)
-        # include = core.CfnInclude(self, 'ExistingInfrastructure',
-        #    template=template,
-        # )
-
-
-class ClusterStack(core.Stack):
-    """Create the Stack and delegate to specific Construct for the creation of all the resources for the Cluster."""
-
-    def __init__(self, scope: core.Construct, construct_id: str, cluster: SlurmCluster, **kwargs) -> None:
-        super().__init__(scope, construct_id, **kwargs)
-        self._cluster = cluster
-
-        # Storage filesystem Ids
-        self._storage_resource_ids = {storage_type: [] for storage_type in SharedStorage.Type}
-
-        # Compute security group Ids
-        # TODO: add sgs created from main stack
-        self._compute_security_groups = []
-        self._compute_security_groups.extend(self._cluster.compute_security_groups)
-
-        HeadNodeConstruct(self, "HeadNode", cluster.head_node)
+    def _add_parallelcluster_hit_policies(self):
```

Comment:
What about renaming hit to Slurm?

We should find a way to separate slurm specific resources/logic by awsbatch one.